docs/developers_guide/deploying_spack.rst

```diff
@@ -0,0 +1,482 @@
+    cd compass/
+    git clone git@github.com:MPAS-Dev/compass.git main
+    cd main/
+    git remote add xylar/compass git@github.com:xylar/compass.git
+
+Now, we need to set up compass and build spack packages making use of the
+updated mache.  This involves changing the mache version in a couple of places
+in compass and updating the version of compass itself to a new alpha, beta or
+rc.  As an example, we will use the branch
+`simplify_local_mache <https://github.com/xylar/compass/tree/simplify_local_mache>`_.
+
+Often, we will need to test with a ``mache`` branch that has changes needed
+by compass.  Here, we will use the ``update_cime_machines`` on the fork
+``xylar/mache`` as an example.
+
+We also need to make sure there is a spack branch for the version of mache.
+This is a branch off of the develop branch on
+`E3SM’s spack repo <https://github.com/E3SM-Project/spack>`_ that has any
+updates to packages required for this version of mache and compass. The
+spack branch will omit any alpha, beta or rc suffix on the mache version
+because it is intended to be the spack branch we will use once the `mache`
+release happens.  In this example, we will work with the branch
+`spack_for_mache_1.12.0 <https://github.com/E3SM-Project/spack/tree/spack_for_mache_1.12.0>`_.
+
+Here's how to get a branch of compass we're testing (``simplify_local_mache``
+in this case) as a local worktree:
+
+.. code-block:: bash
+
+    # get my branch
+    git fetch --all -p
+    # make a worktree for checking out my branch
+    git worktree add ../simplify_local_mache -b simplify_local_mache \
+        --checkout xylar/compass/simplify_local_mache
+    cd ../simplify_local_mache/
+
+You will also need a local installation of
+`Mambaforge <https://github.com/conda-forge/miniforge#mambaforge>`_.
+Compass can do this for you if you haven't already installed it.  If you want
+to download it manually, use the Linux x86_64 version for all our supported
+machines.
+
+.. note::
+
+    We have found that an existing Miniconda3 installation **does not** always
+    work well for compass, so please start with Mambaforge instead.
+
+.. note::
+
+  You definitely need your own local Mambaforge installation -- you can’t use
+  a system version or a shared one like E3SM-Unified.
+
+Define a location where Mambaforge is installed or where you want to install
+it:
+
+.. code-block:: bash
+
+    # change to your conda installation
+    export CONDA_BASE=${HOME}/mambaforge
+
+Okay, we're finally ready to do a test spack build for compass.
+To do this, we call the ``configure_compass_env.py`` script using
+``--mache_fork``, ``--mache_branch``, ``--update_spack``, ``--spack`` and
+``--tmpdir``. Here is an example appropriate for Anvil or Chrysalis:
+
+.. code-block:: bash
+
+    export TMPDIR=/lcrc/group/e3sm/${USER}/spack_temp
+    ./conda/configure_compass_env.py \
+        --conda ${CONDA_BASE} \
+        --mache_fork xylar/mache \
+        --mache_branch update_cime_machines \
```

Comment:
I ran into the same thing

Reply:
Misery loves company!

Reply:
And now it works for me as well on compy

Reply:
Is there another way to clarify this?  Maybe:
```suggestion
        --mache_fork <fork> \
        --mache_branch <branch> \
```
I worry that having the main branch as the example might be an especially poor choice because that will actually work if someone just blindly copy/pastes the example without realizing that it needs to be a specific fork/branch for the given project.

Reply:
That works for me.

Reply:
Done.